jtwig-core/src/main/java/org/jtwig/render/stream/RenderTask.java

```diff
@@ -35,9 +35,11 @@
     public void run() {
         try {
             content.render(context);
-            context.renderStream().close().merge().notifyTaskFinished();
+            context.renderStream().close().merge();
         } catch (Exception e) {
             logger.error("[Concurrent Beta] Rendering failed", e);
+        } finally {
+            context.renderStream().notifyTaskFinished();
```

Comment:
This one had been giving me a headache for a bit, so I finally ran it down. I'm not sure that this one catch nails everything, but it's a start.